stan/math/prim/mat/fun/scale_matrix_exp_multiply.hpp

```diff
@@ -27,6 +31,31 @@ inline Eigen::Matrix<double, -1, Cb> scale_matrix_exp_multiply(
   return matrix_exp_action_handler().action(A, B, t);
 }
 
+/**
+ * Return product of exp(At) and B, where A is a NxN matrix,
+ * B is a NxCb matrix and t is a scalar.
+ *
+ * Generic implementation when arguments are not double.
+ *
+ * @tparam Ta scalar type matrix A
+ * @tparam Tb scalar type matrix B
+ * @tparam Cb Columns matrix B
+ * @param[in] A Matrix
+ * @param[in] B Matrix
+ * @param[in] t double
+ * @return exponential of At multiplies B
+ */
+template <typename Tt, typename Ta, typename Tb, int Cb>
+inline Eigen::Matrix<stan::return_type_t<Tt, Ta, Tb>, -1, Cb>
+scale_matrix_exp_multiply(const Tt& t, const Eigen::Matrix<Ta, -1, -1>& A,
+                          const Eigen::Matrix<Tb, -1, Cb>& B) {
+  check_nonzero_size("scale_matrix_exp_multiply", "input matrix", A);
+  check_nonzero_size("scale_matrix_exp_multiply", "input matrix", B);
```

Comment:
This should be replaced by:
```
  if (A.size() == 0 && B.size() == 0)
    return {};
```

To trigger that, add the following (or an equivalent code as you see fit) to `test/unit/math/mix/mat/fun/scale_matrix_exp_multiply_test.cpp` (I have added that to `test/unit/math/rev/mat/fun/scale_matrix_exp_multiply_test.cpp`  which you are removing):
```
TEST(MathMatrix, matrix_exp_multiply_0x0) {
  using stan::math::var;
  const double t = 1.0;
  Eigen::Matrix<var, -1, -1> A(0, 0);
  Eigen::Matrix<var, -1, -1> B(0, 0);
  EXPECT_EQ(stan::math::scale_matrix_exp_multiply(t, A, B).size(), 0);
}
```

Reply:
Thanks.  I put the conditional after `check_multiplicable` and `test_square` to make sure that everything's the right shape even if it's size zero.  There was already a full test for 0x0 times 0x0.